### PR TITLE
DEVENGAGE-1542 Adding Knowledge v2 resources 

### PR DESCRIPTION
**Pull request by diemenschmaschine92**
Knowledge v2 have been rolled out, but v1 is still in use by some orgs.  This adds the v2 resources.  Since the test org is using v2, the v1 tests will no longer pass, so skipping those.  The Knowledge API results in some unorthodox code for Terraform, so if somethings seems unusual just ask and I should be able to explain.

### CODE REVIEW

File: genesyscloud/data_source_genesyscloud_knowledge_category.go [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+		for pageNum := 1; ; pageNum++ {
+			const pageSize = 100
+			publishedKnowledgeBases, _, getPublishedErr := knowledgeAPI.GetKnowledgeKnowledgebases("", "", "", fmt.Sprintf("%v", pageSize), knowledgeBaseName, "", true, "", "")
+			unpublishedKnowledgeBases, _, getUnpublishedErr := knowledgeAPI.GetKnowledgeKnowledgebases("", "", "", fmt.Sprintf("%v", pageSize), knowledgeBaseName, "", false, "", "")
+
+			if getPublishedErr != nil {
+				return resource.NonRetryableError(fmt.Errorf("Failed to get knowledge base %s: %s", knowledgeBaseName, getPublishedErr))
+			}
+			if getUnpublishedErr != nil {
+				return resource.NonRetryableError(fmt.Errorf("Failed to get knowledge base %s: %s", knowledgeBaseName, getPublishedErr))
```
Comment (charliecon) [inline genesyscloud/data_source_genesyscloud_knowledge_category.go:50]: Should you be returning `getUnpublishedErr` here?

Comment (diemenschmaschine92) [reply genesyscloud/data_source_genesyscloud_knowledge_category.go]: Yes, thanks.

File: genesyscloud/resource_genesyscloud_knowledge_v1_document_test.go [hunk 2]
```diff
@@ -0,0 +1,241 @@
...
+}
+
+func generateFaq(question string, answer string, alternatives []string) string {
+	formattedAlternatives := ""
+	for i, alternative := range alternatives {
+		if i > 0 {
+			formattedAlternatives += ", "
+		}
+		formattedAlternatives += strconv.Quote(alternative)
+	}
```
Comment (charliecon) [inline genesyscloud/resource_genesyscloud_knowledge_v1_document_test.go:179]: It's easier if you pass in an array of quoted strings and say `fmt.Sprintf("x = [%s]", strings.Join(alternatives, ", "))`

File: genesyscloud/resource_genesyscloud_knowledge_v1_document_test.go [hunk 3]
```diff
@@ -0,0 +1,241 @@
...
+}
+
+func generateArticle(title string, contentLocationUrl string, alternatives []string) string {
+	formattedAlternatives := ""
+	for i, alternative := range alternatives {
+		if i > 0 {
+			formattedAlternatives += ", "
+		}
+		formattedAlternatives += strconv.Quote(alternative)
+	}
```
Comment (charliecon) [inline genesyscloud/resource_genesyscloud_knowledge_v1_document_test.go:199]: Same here

**Review by carnellj-genesys — APPROVED** [review]

**Review by charliecon — APPROVED** [review]

**Comment by diemenschmaschine92** [comment]
@charliecon Ok to merge?

**Comment by charliecon** [comment]
> @charliecon Ok to merge?

@diemenschmaschine92  Yeah go for it 👍🏻

**Pull request merged**